src/client/linter.ts

```diff
@@ -0,0 +1,91 @@
+import {
+	TextDocument,
+	Position,
+	Range,
+	DiagnosticSeverity,
+	Diagnostic,
+	languages,
+	workspace,
+	Uri
+} from "vscode";
+import { tmpdir } from "os";
+import { sep } from "path";
+import { arrayInclude, trimBoth, getCwd } from "./utils";
+import { execV } from "./exec";
+import { resolve } from "path";
+
+const outDir = `${tmpdir()}${sep}vscode_vlang${sep}`;
+
+export interface ErrorInfo {
+	file: string;
+	line: number;
+	column: number;
+	message: string;
+	stderr: string;
+}
+
+export const collection = languages.createDiagnosticCollection("V");
+
+export function lint(document: TextDocument): boolean {
+	// Don't lint files that are not in the workspace
+	if (!workspace.getWorkspaceFolder(document.uri)) {
+		return false;
+	}
+
+	const cwd = getCwd();
+	// const filename = relative(cwd, document.uri.fsPath);
+
+	const cmd = `-o ${outDir}lint.c .`;
```

Comment:
This also produces invalid errors, e.g.:
![image](https://user-images.githubusercontent.com/1119267/72203646-6ea4b380-347f-11ea-8bb5-cca9d297c431.png)

I changed `.` to path to file, and now actual errors are shown (see my screenshot in comments).

Reply:
the problem when `v -o /tmp/vscode_vlang/linter.c currentActiveFile.v` is the error is invalid
for example:
a.v
```v
struct Foo {}
```
b.v
```v 
struct Bar {}
```
sometimes the error said: `Foo` is not exist.

Reply:
Aah, nevermind, it's okay.

Reply:
![](https://user-images.githubusercontent.com/1119267/72203487-3b612500-347d-11ea-8f33-de3a1c53bd56.png)
But this is still a bug  😁

Reply:
This error was from an invalid file that was not supposed to be in the workspace. Aside wrong parsing, it's not an issue.

Reply:
Yeah, I forgot that `path` info